### ISSUE TITLE
Upgrade to MathSAT 5.5.0

### ISSUE THREAD

**Issue by marcogario** [pysmt/pysmt#409]

**Comment by marcogario** [comment]
See MathSAT 5.5.0 http://mathsat.fbk.eu/download.html

**Comment by marcogario** [comment]
Preliminary testing with the pre-release shows that the upgrade should not require changes to pySMT! 🎉

**Comment by mikand** [comment]
AFAIK @agriggio is going to make a complete release (for all the platforms) of 5.5.0 soon (next week?), so we shall wait. Also, this release contains an incomplete non-linear solver that we can exploit!

**Comment by mikand** [comment]
One point that is very relevant though, is whether or not this new release will be compatible with Ubuntu 14.04 where our tests are running.

@agriggio are you planning to include an `.so` for Linux or just the `.a`?

**Comment by agriggio** [comment]
I'll do what it takes to make it work... just let me know what you need

**Comment by mikand** [comment]
So, the previous release (5.4.1) was working fine for Ubuntu 16.04+, but on Ubuntu 14.04 with default GCC compiler we got:
```
...
creating build/lib.linux-x86_64-2.7
c++ -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-z,relro -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -D_FORTIFY_SOURCE=2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wl,-z,relro -D_FORTIFY_SOURCE=2 -g -fstack-protector-strong -Wformat -Werror=format-security build/temp.linux-x86_64-2.7/mathsat_python_wrap.o -L../lib -lmathsat -lgmpxx -lgmp -o build/lib.linux-x86_64-2.7/_mathsat.so
/usr/bin/ld: ../lib/libmathsat.a(fmelim.cpp.o): unrecognized relocation (0x2a) in section `.text'
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
error: command 'c++' failed with exit status 1
```

You can see the full log on Travis: https://travis-ci.org/pysmt/pysmt/jobs/246006483#L998

Clearly this is a linker issue that seems related to  https://bugs.launchpad.net/ubuntu/+source/binutils/+bug/1545653.

I remember you suggested to include a shared library for the linux version analogously to the dll distributed for windows. This should enhance compatibility among GCC versions used to distribute mathsat and to compile the bindings.

Of course,if you have any other solution (e.g. compile MathSAT on 14.04 for distribution) to make things work under 14.04 we are fine, but upgrading to 16.04 is quite problematic on Travis right now...

**Comment by marcogario** [comment]
Fixed in #453
